Car.cpp

```diff
@@ -1,7 +1,41 @@
 #include "Car.hpp"
 #include <iostream>
 
-void Car::turnLeft()      { std::cout << __FUNCTION__ << std::endl; }
-void Car::turnRight()     { std::cout << __FUNCTION__ << std::endl; }
-void Car::brake()         { std::cout << __FUNCTION__ << std::endl; }
-void Car::accelerate(int) { std::cout << __FUNCTION__ << std::endl; }
+void Car::turnLeft()
+{
+    if(velocity!=0)
+        std::cout << __FUNCTION__ << std::endl;
+    else
+        std::cout << "You must move to turn!" << std::endl;
+}
+void Car::turnRight()
+{
+    if(velocity!=0)
+        std::cout << __FUNCTION__ << std::endl;
+    else
+        std::cout << "You must move to turn!" << std::endl;
+}
+void Car::accelerate(int speed)
+{ 
+    std::cout << __FUNCTION__ << std::endl;
+    if (velocity += speed < -30)
+        Car::velocity = -30;
+    else if(speed<Car::Vmax)
+        Car::velocity += speed;
+    else 
+        Car::velocity = Car::Vmax;
+    
+    std::cout << "Current speed:" << Car::velocity << std::endl;
+
+}
+void Car::brake(int b)
+{
+    std::cout << __FUNCTION__ << std::endl;
+    if(Car::velocity - b*15 <0)

```

Comment:
skad jest to "15"?

Reply:
tak samo z głowy, założyłem, ze w ciągu jednej sekundy hamowania, samochód zwalnia o 15.